src/components/ui/drawer.tsx

```diff
@@ -0,0 +1,107 @@
+'use client';
+import { forwardRef } from 'react';
+import { Drawer as DrawerPrimitive } from 'vaul';
+import { cn } from '@/lib/utils';
+
+const Drawer = ({
+  shouldScaleBackground = true,
+  ...props
+}: React.ComponentProps<typeof DrawerPrimitive.Root>) => (
+  <DrawerPrimitive.Root
+    shouldScaleBackground={shouldScaleBackground}
+    {...props}
+  />
+);
+Drawer.displayName = 'Drawer';
+
+const DrawerTrigger = DrawerPrimitive.Trigger;
+
+const DrawerPortal = DrawerPrimitive.Portal;
+
+const DrawerClose = DrawerPrimitive.Close;
+
+const DrawerOverlay = forwardRef<
+  React.ElementRef<typeof DrawerPrimitive.Overlay>,
+  React.ComponentPropsWithoutRef<typeof DrawerPrimitive.Overlay>
+>(({ className, ...props }, ref) => (
+  <DrawerPrimitive.Overlay
+    ref={ref}
+    className={cn('fixed inset-0 z-50 bg-overlay', className)}
+    {...props}
+  />
+));
+DrawerOverlay.displayName = DrawerPrimitive.Overlay.displayName;
+
+const DrawerContent = forwardRef<
+  React.ElementRef<typeof DrawerPrimitive.Content>,
+  React.ComponentPropsWithoutRef<typeof DrawerPrimitive.Content>
+>(({ className, children, ...props }, ref) => (
+  <DrawerPortal>
+    <DrawerOverlay />
+    <DrawerPrimitive.Content
+      ref={ref}
+      className={cn(
+        'fixed inset-x-0 bottom-0 z-50 mt-24 flex h-auto flex-col rounded-t-xl border bg-background',
+        className,
+      )}
+      {...props}
+    >
+      <div className="mx-auto mt-4 h-2 w-[100px] rounded-full bg-primary-lightest" />
```

Comment:
I think this is a mistake, but it looks like the margin-top is 17px according to Figma.
Could you confirm this with the designers?

<img width="543" alt="image" src="https://github.com/Tascurator/tascurator-frontend/assets/124953279/47942dcb-7483-424b-a640-e783a6d7d761">

Reply:
> I think this is a mistake, but it looks like the margin-top is 17px according to Figma. Could you confirm this with the designers?
> 
> <img alt="image" width="543" src="https://private-user-images.githubusercontent.com/124953279/317953669-47942dcb-7483-424b-a640-e783a6d7d761.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIwMjMxMzksIm5iZiI6MTcxMjAyMjgzOSwicGF0aCI6Ii8xMjQ5NTMyNzkvMzE3OTUzNjY5LTQ3OTQyZGNiLTc0ODMtNDI0Yi1hNjQwLWU3ODNhNmQ3ZDc2MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMlQwMTUzNTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zM2I0MjA1NTAxNzA1MjVkNTY0OTM5MmNjZmEwNGU3ODVhNzJmYWViNDFjNzVmMGFkZWNhNTZiNGE5ODc4YjExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.hdHpFPWhfvIG4l5ILrMrk1_r0B-EVXc9YyXyIFsQWLY">

I have changed the margin-top to 16px.